src/with_clause_parser.c

```diff
@@ -161,11 +164,9 @@ parse_arg(WithClauseDefinition arg, DefElem *def)
 
 	Assert(OidIsValid(in_fn));
 
-	PG_TRY();
-	{
-		val = OidInputFunctionCall(in_fn, value, typIOParam, -1);
-	}
-	PG_CATCH();
+	fmgr_info(in_fn, &flinfo);
+
+	if (!InputFunctionCallSafe(&flinfo, value, typIOParam, -1, (Node *) &escontext, &val))
```

Comment:
I think this function only exists in pg15+.

Reply:
Worse, PG16+.

```
mats@fury:~/repos/postgres$ git tag --contains d9f7f5d32f201bec61fef8104aafcb77cecb4dcb
REL_16_0
REL_16_1
REL_16_2
REL_16_3
REL_16_4
REL_16_5
REL_16_6
REL_16_BETA1
REL_16_BETA2
REL_16_BETA3
REL_16_RC1
REL_17_0
REL_17_1
REL_17_2
REL_17_BETA1
REL_17_BETA2
REL_17_BETA3
REL_17_RC1
```

Reply:
Went back to using the PG_TRY + PG_CATCH but checking the category and only allow data exception category errors. These are syntax errors, and some more, that are thrown by input functions.